lib/matplotlib/legend.py

```diff
@@ -480,6 +482,11 @@ def val_or_rc(val, rc_name):
         self._mode = mode
         self.set_bbox_to_anchor(bbox_to_anchor, bbox_transform)
 
+        # Figure out if self.shadow is valid
+
+        if not (is_color_like(self.shadow) or isinstance(self.shadow, bool)):
+            raise ValueError('Legend shadow must be a valid color or bool')
```

Comment:
We try to tell the user not only that their value is wrong, but what value they actually passed us:

```suggestion
            raise ValueError(
                'Legend shadow must be a valid color or bool, not '
                f'{self.shadow!r} of type {type(self.shadow)}.'
            )
```